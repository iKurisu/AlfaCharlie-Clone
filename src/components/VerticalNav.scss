@import 'sass/_variables.scss';

.vertical-nav {
  width: 30px;
  height: 80vh;
  position: fixed;
  right: 46px;
  top: 10vh;
  z-index: 10;
}

main > .vertical-nav {
  display: flex;
  align-items: center;
  justify-content: center;
}

.vertical-nav-links {
  transform: rotate(90deg);
  white-space: nowrap;
  line-height: 15px;
}

@media (max-width: $l) {
  .vertical-nav {
    display: none;
  }
}

@media (min-width: $xxl) {
  .vertical-nav {
    right: 57px;
  }
}
