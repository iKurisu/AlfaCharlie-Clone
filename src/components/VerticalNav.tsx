import React from "react";
import "./VerticalNav.scss";

interface NavProps {
  links: [string, boolean][];
  location?: string;
}

const VerticalNav = ({ links }: NavProps): JSX.Element => (
  <nav className="vertical-nav">
    <ul className="vertical-nav-links">
      {links.map(
        ([link, active], id): JSX.Element => (
          <li key={id}>
            <span className={`link-wrapper${active ? " active" : ""}`}>
              <span className="link-name">{link}</span>
            </span>
          </li>
        )
      )}
    </ul>
  </nav>
);

export default VerticalNav;
