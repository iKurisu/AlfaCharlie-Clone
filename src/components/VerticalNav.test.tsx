import React from "react";
import renderer from "react-test-renderer";
import VerticalNav from "./VerticalNav";

it("renders correctly", (): void => {
  const links: string[] = ["Expertise", "Team", "Clients"];
  const component = renderer.create(<VerticalNav links={links} />);

  expect(component).toMatchSnapshot();
});
