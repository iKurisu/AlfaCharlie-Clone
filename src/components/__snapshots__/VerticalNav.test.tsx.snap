// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="nav"
>
  <ul>
    <li>
      <span
        className="link-wrapper active"
      >
        <span
          className="link-name"
        >
          E
        </span>
      </span>
    </li>
    <li>
      <span
        className="link-wrapper active"
      >
        <span
          className="link-name"
        >
          T
        </span>
      </span>
    </li>
    <li>
      <span
        className="link-wrapper active"
      >
        <span
          className="link-name"
        >
          C
        </span>
      </span>
    </li>
  </ul>
</nav>
`;
